src/annotations/items/text/TextCursor.cpp

```diff
@@ -99,6 +140,29 @@ void TextCursor::moveCursorForwardBy(const QString &text, int moveBy)
     }
 }
 
+void TextCursor::moveCursorAtPreviousWordBeginning(const QString &text)
+{
+    int lastNonSpacePos{-1};
+    int pos{mPosition-1};
+
+    while (pos >= 0) {
+        if (text.at(pos).isSpace()) {
+            if (lastNonSpacePos >= 0) {
+                break;
+            }
+        } else {
+            lastNonSpacePos = pos;
+        }
+        -- pos;
+    }
+
+    if (lastNonSpacePos >= 0) {
+        mPosition = lastNonSpacePos;
+    } else {
+        mPosition = 0;
```

Comment:
In the `moveCursorAtNextWordBeginning` you don't have this case where you move the cursor to the end when no space was found, maybe add a case for it?

Reply:
In the `moveCursorAtNextWordBeginning` method, if no space is found, `mPosition` will end up being at the last position.